libecole/src/reward/solvingtime.cpp

```diff
@@ -0,0 +1,55 @@
+#include <chrono>
+
+#include "ecole/reward/solvingtime.hpp"
+#include "ecole/scip/model.hpp"
+
+namespace ecole::reward {
+
+const int MUS_IN_SECONDS = 1000000;
+
+static auto process_solving_time(scip::Model const& model) {
+	switch (model.get_stage()) {
+	// Only stages when the following call is authorized
+	case SCIP_STAGE_PROBLEM:
+	case SCIP_STAGE_TRANSFORMING:
+	case SCIP_STAGE_TRANSFORMED:
+	case SCIP_STAGE_INITPRESOLVE:
+	case SCIP_STAGE_PRESOLVING:
+	case SCIP_STAGE_EXITPRESOLVE:
+	case SCIP_STAGE_PRESOLVED:
+	case SCIP_STAGE_INITSOLVE:
+	case SCIP_STAGE_SOLVING:
+	case SCIP_STAGE_SOLVED:
+		return SCIPgetSolvingTime(model.get_scip_ptr());
+	default:
+		return decltype(SCIPgetSolvingTime(nullptr)){0};
+	}
+}
+
+void SolvingTime::reset(scip::Model& model) {
+	model.set_params({{"timing/clocktype", 1}});
+
+	if (wall) {
+		auto now_in_mus = std::chrono::duration_cast<std::chrono::microseconds>(
+			std::chrono::high_resolution_clock::now().time_since_epoch());
```

Comment:
Note from https://en.cppreference.com/w/cpp/chrono/high_resolution_clock

> The high_resolution_clock is not implemented consistently across different standard library implementations, and its use should be avoided. It is often just an alias for std::chrono::steady_clock or std::chrono::system_clock, but which one it is depends on the library or configuration.

Reply:
Switched to `steady_clock`.